core/src/test/java/org/elasticsearch/action/support/master/IndexingMasterFailoverIT.java

```diff
@@ -73,9 +73,10 @@ public void testMasterFailoverDuringIndexingWithMappingChanges() throws Throwabl
                 .put("discovery.zen.join_timeout", "10s")  // still long to induce failures but to long so test won't time out
                 .put(DiscoverySettings.PUBLISH_TIMEOUT_SETTING.getKey(), "1s") // <-- for hitting simulated network failures quickly
                 .put(ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING.getKey(), 2)
+                .put(DiscoverySettings.INITIAL_STATE_TIMEOUT_SETTING.getKey(), "0")  // <-- we wait for cluster formation at the end
```

Comment:
It exposes the sequential nature that is used by `startNodes` to users of this API, which is completely unnecessary (I've even outlined a solution) and the only reason requiring this setting. Calling `startNodes(3)` should not trip up because of some internals require an extra setting to be passed. Let's have a clean API that does the right thing.